<launch>
 <group ns="xtlidar">
  <node name="xtlidar" pkg="xtlidar_driver" type="xtlidar_driver_node" output="screen" >
    <param name="type" value="uart" />
    <param name="platform" value="LDS-50C-2" />
    <param name="port" value="/dev/xtark_lidar" />
    <param name="baud_rate" value="768000" />
    <param name="max_dist" value="25.0"/>
    <rosparam param="rate_list">[768000, 1000000 ]</rosparam>
    <param name="raw_bytes" value="3"/>
    <param name="frame_id" value="laser" />
    <param name="output_360" value="true" />
    <param name="output_scan" value="true" />
    <param name="output_cloud" value="true"/>
    <param name="with_angle_filter" value="true"/>
    <param name="min_angle" value="-2.1"/>
    <param name="max_angle" value="2.1"/>
    <param name="hard_resample" value="true"/>
    <param name="soft_resample" value="false"/>
    <param name="resample_res" value="0.25"/>
    <param name="unit_is_mm" value="true"/>
    <param name="with_confidence" value="true"/>
    <param name="with_checksum" value="true"/>
    <remap from="/xtlidar/scan" to="/scan"/>
  </node>
 </group>
</launch>

